File: playwright-csharp/tests/Pages/RestfulBooker/Admin/RoomsPage.cs
using Microsoft.Playwright;
using TestTracks.Playwright.CSharp.Configuration;
using TestTracks.Playwright.CSharp.UI.Common;

namespace TestTracks.Playwright.CSharp.Specs.Pages.RestfulBooker.Admin;

public sealed class RoomsPage : BasePage
{
    private ILocator RoomNameText(string roomName) =>
        Page.GetByText(roomName, new() { Exact = true });

    public RoomsPage(IPage page, TestSettings settings)
        : base(page, settings)
    {
    }

    public ILocator RoomNamed(string roomName)
    {
        return RoomNameText(roomName);
    }
}

File: playwright-java/src/test/java/testtracks/pages/restfulbooker/admin/RoomsPage.java
package testtracks.pages.restfulbooker.admin;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import testtracks.configuration.TestSettings;
import testtracks.ui.common.BasePage;

public final class RoomsPage extends BasePage {
  public Locator roomNamed(String roomName) {
    return page.getByText(roomName, new Page.GetByTextOptions().setExact(true));
  }

  public RoomsPage(Page page, TestSettings settings) {
    super(page, settings);
  }
}
